Core/FitParameter.hpp

```diff
@@ -1,230 +1,67 @@
-  /// operator<<().
-  virtual std::string to_str() const;
-
-  /// String with detailed information about the parameter. Used in
-  /// operator<<().
-  virtual std::string val_to_str() const;
-
-protected:
-  virtual std::string className() const { return "Double"; }
-
-  /// Are valid bounds defined for this parameter?
-  bool HasBounds;
-
-  /// Do you want to keep parameter fixed?
-  bool IsFixed;
-
-  /// Parameter value
-  double Value;
-
-  /// Parameter bounds
-  std::pair<double, double> Bounds;
-
-  /// No error / symmetric error / asymmetric error
-  ErrorType ErrType;
-
-  /// Lower parameter error
-  std::pair<double, double> Error;
-
-  virtual void SetErrorType(ErrorType t) { ErrType = t; }
-
-  /// Check if \p min and \p max are valid bounds
-  bool check_bounds(const std::pair<double, double> bounds) const;
-
-private:
-  friend class boost::serialization::access;
-  template <class archive>
-  void serialize(archive &ar, const unsigned int version) {
-    using namespace boost::serialization;
-    //    ar &boost::serialization::make_nvp(
-    //        "Parameter", boost::serialization::base_object<Parameter>(*this));
-    ar &make_nvp("Name", Name);
-    ar &make_nvp("Bounds", Bounds);
-    ar &make_nvp("Fix", IsFixed);
-    ar &make_nvp("Value", Value);
-    ar &make_nvp("Bounds", Bounds);
-    try {
-      ar &make_nvp("ErrorType", ErrType);
-      ar &make_nvp("Error", Error);
-    } catch (...) {
-      Error = std::pair<double, double>(0, 0);
-      ErrType = ErrorType::SYM;
+template <typename T> struct FitParameter {
+  FitParameter() = default;
+  FitParameter(std::string name, T val, bool isfixed = true)
+      : HasBounds(false), IsFixed(isfixed), Value(val), Name(name),
+        Bounds(0.0, 0.0) {}
+  FitParameter(std::string name, T val, T min, T max, bool isfixed = true)
+      : HasBounds(true), IsFixed(isfixed), Value(val), Name(name),
+        Bounds(min, max) {}
+  bool HasBounds = false;
+  bool IsFixed = false;
+  T Value;
+  std::pair<T, T> Error;
+  std::string Name = "";
+  std::pair<T, T> Bounds;
+
+  friend std::ostream &operator<<(std::ostream &os,
+                                  const FitParameter<double> &x) {
+    os << x.Name << ": " << x.Value;
+    if (x.Error.first != 0.0 || x.Error.second != 0.0) {
+      if (x.Error.first == x.Error.second) {
+        os << " +- " << x.Error.second;
+      } else {
+        os << " + " << x.Error.second << " - " << x.Error.first;
+      }
+      if (x.HasBounds) {
+        os << " Bounds: [" << x.Bounds.first << ", " << x.Bounds.second << "]";
+      }
     }
+    if (x.IsFixed)
+      os << " (fixed)";
+    return os;
   }
 };
-BOOST_SERIALIZATION_SHARED_PTR(ComPWA::FitParameter)
+
+using FitParameterList = std::vector<FitParameter<double>>;
 
 } // namespace ComPWA
 
+namespace boost {
+namespace serialization {
+
+template <class Archive>
+void serialize(Archive &ar, ComPWA::FitParameter<double> &Par,
```

Comment:
If FitParameter is template class, then it will be better the serialize also support the template